docs/Configuring a Harmony service.ipynb

```diff
@@ -0,0 +1,577 @@
+    "            \"--harmony-input\",\n",
+    "            \"{{inputs.parameters.operation}}\",\n",
+    "            \"--harmony-sources\",\n",
+    "            \"{{inputs.artifacts.metadata.path}}/{{inputs.parameters.stac-catalog-link}}\",\n",
+    "            \"--harmony-metadata-dir\",\n",
+    "            \"{{outputs.artifacts.metadata.path}}\"\n",
+    "          ]\n",
+    "        envFrom:\n",
+    "          - configMapRef:\n",
+    "              name: harmony-env\n",
+    "          - secretRef:\n",
+    "              name: harmony-secrets\n",
+    "        env:\n",
+    "          - name: STAGING_PATH\n",
+    "            value: \"{{inputs.parameters.STAGING_PATH}}\"\n",
+    "          - name: APP_NAME\n",
+    "            value: \"your-service-name\"\n",
+    "```\n",
+    "\n",
+    "In this example the majority of the YAML can be copied. There are currently (as of 2021-04-21) 6 values that should be updated from this example:\n",
+    "\n",
+    "* `metadata.name`: This is the overall name for your service. Most service names have so far followed the convention: `teamname-service-name`, for example `sds-maskfill` or `asf-gdal-subsetter`.\n",
+    "* `spec.entrypoint`: This value should match the `name` property of the first list item shown in the `spec.templates` section.\n",
+    "* `spec.templates`, in the `steps` template, `name`: This value should match the `spec.entrypoint` value. Each step is a service that Harmony invokes. It is possible to invoke multiple services, via chaining, by having multiple steps. An example of this will be shown below.\n",
+    "* `spec.templates`, in the `steps` template, `steps` list, `last-item`, `template`: This value specifies the service template to use for this step. It should match the `name` of the `services` list element in the `spec.templates` list.\n",
+    "* `spec.template`, in the `services` template, `name`: This should be a descriptive name of the service being called. The values within this template tell Argo the Docker image to build a container from, the arguments to supply when invoking this container, and the environment variables to supply.\n",
+    "* `spec.template`. in the `services` template, `container.env`, `APP_NAME`: The `value` of this environment variable will be used in several places. For example, this string value will be contained in the logging output from Harmony and will enable queries to be made for log messages from specific services.\n",
+    "\n",
+    "The example template above has the strings \"your-service-name\", \"your-service-name-steps\" and \"your-service-name-service\" in the six places described. If this template were for the `sds-maskfill` service, those values would be: \"sds-maskfill\", \"sds-maskfill-steps\" and \"sds-maskfill-service\", as can be seen [here](https://github.com/nasa/harmony/blob/main/config/workflow-templates/sds-maskfill.yaml).\n",
+    "\n",
+    "**A chained Harmony request:**\n",
```

Comment:
This isn't really a suggestion, just an observation: A chained workflow use case is more tied to the supported collection(s), unless the service doesn't operate as a standalone for some reason. So it's just something to keep in mind, as this is really helpful for providers adding new collection support but maybe not as relevant for a new service configuration.